src/python/pants/backend/shell/util_rules/shell_command.py

```diff
@@ -111,11 +113,11 @@ class ShellCommandProcessFromTargetRequest:
 async def _prepare_process_request_from_target(shell_command: Target) -> ShellCommandProcessRequest:
     description = f"the `{shell_command.alias}` at `{shell_command.address}`"
 
-    interactive = shell_command.has_field(ShellCommandRunWorkdirField)
-    if interactive:
-        working_directory = shell_command[ShellCommandRunWorkdirField].value or ""
-    else:
-        working_directory = shell_command.address.spec_path
+    interactive = shell_command.has_field(ShellCommandIsInteractiveField)
+    working_directory = shell_command[ShellCommandWorkdirField].value
+
+    if interactive and working_directory is None:
+        working_directory = "."
```

Comment:
Is `.` here the build root? Or the dir of the target?

Reply:
It's the build root.

Reply:
It is now the dir of the target :)